src/extensions/step_handlers/wim_handler/src/wim_step_handler.cpp

```diff
@@ -39,6 +41,76 @@ static std::string make_preferred(const char* path, const char* file = nullptr)
     return preferred;
 }
 
+/**
+ * @brief Verifies that the file path exists.
+ *
+ * @param payloadFilePath The path to the file.
+ * @return HRESULT S_OK when the file exists; an error hresult, otherwise.
+ */
+static HRESULT EnsurePayloadExists(const std::string& payloadFilePath)
+{
+    if (ADUCPAL_access(payloadFilePath.c_str(), F_OK) != 0)
+    {
+        int err;
+        _get_errno(&err);
+        return HRESULT_FROM_WIN32(err);
+    }
+
+    return S_OK;
+}
+
+/**
+ * @brief Get the volume label for the drive
+ *
+ * @param[in] driveRoot The drive root path name, e.g. "C:\\" or "D:\\"
+ * @param[out] outLabel The pointer to an existing std::string object to receive the resulting volume label.
+ * @return HRESULT The value of GetLastError() as processed by HRESULT_FROM_WIN32 macro on error, or S_OK on success.
+ * @details For failed returned HRESULT, the std::string object pointed to by outLabel will not be affected.
+ */
+static HRESULT GetDriveLabel(const std::string& driveRoot, std::string* outLabel)
```

Comment:
seeing how line 86 can throw, consider throwing on error and returning a string, e.g.

``` C++
std::string GetDriveLabel(root) {
if (!GetVolumeInformation...)
{
  throw HRESULT_FROM_WIN32(GetLAstError();
}

return volumename;